test/unit/interface/test_cholesky.cpp

```diff
@@ -0,0 +1,303 @@
+//
```

Comment:
Not sure... if we want to use gtest I think we need c++.

Reply:
I think here using `.c` would be more useful than using GTest, since we want to test that everything work as expected when using the interface, while correctness tests are already handled extensively elsewhere.

Reply:
Yeah, obviously things like GTest may need a C++ compiler. However, I'd say even just making a dummy `.c` file which does something silly like:
```
void my_cholesky_d(char uplo, double* a, int m, int n, int mb, int nb, int lld,
                           const MPI_Comm& communicator, int nprow, int npcol) {
    cholesky_d(uplo, a, m, n, mb, nb, lld, communicator, nprow, npcol);
}
```
and then calling `my_cholesky_d` from a C++ file would be better in my opinion just to make sure the C interface can be used in a C file. Note, there may be a better way of doing this, the above is just an example.

Reply:
`Everything works` doesn't mean `everything works correctly`.
A small typo in one when setting one of the sizes in the interface is enough to break everything silently.

Reply:
Yes, I was not suggesting not to test for correctness, but to have a smaller set of C tests with less automation than provided by GTest. Something in the lines of what @msimberg suggests seems to combine the best of both worlds, I'll look into it.

Reply:
I think this is fixed now, I followed @msimberg suggestion of having a dummy `.c` file and then used GTest for the actual tests (with most of the code shared with the standard tests).